File: j-LIB-core/src/main/java/io/github/dot166/jlib/utils/AppUtils.java
package io.github.dot166.jlib.utils;

import android.content.Context;
import android.content.pm.ApplicationInfo;
import android.graphics.drawable.Drawable;

import androidx.appcompat.content.res.AppCompatResources;

public class AppUtils {

    public static Drawable getAppIcon(Context context) {
        ApplicationInfo applicationInfo = context.getApplicationInfo();
        int iconId = applicationInfo.icon;
        return iconId == 0 ? AppCompatResources.getDrawable(context, android.R.drawable.sym_def_app_icon) : AppCompatResources.getDrawable(context, iconId);
    }

    public static String getAppLabel(Context context) {
        ApplicationInfo applicationInfo = context.getApplicationInfo();
        int stringId = applicationInfo.labelRes;
        return stringId == 0 ? applicationInfo.nonLocalizedLabel.toString() : context.getString(stringId);
    }
}

File: j-LIB-core/src/main/java/io/github/dot166/jlib/utils/AppUtils.kt
package io.github.dot166.jlib.utils

import android.content.Context
import android.graphics.drawable.Drawable
import androidx.appcompat.content.res.AppCompatResources

object AppUtils {
    @JvmStatic
    fun getAppIcon(context: Context): Drawable? {
        val applicationInfo = context.applicationInfo
        val iconId = applicationInfo.icon
        return if (iconId == 0) AppCompatResources.getDrawable(
            context,
            android.R.drawable.sym_def_app_icon
        ) else AppCompatResources.getDrawable(context, iconId)
    }

    @JvmStatic
    fun getAppLabel(context: Context): String {
        val applicationInfo = context.applicationInfo
        val stringId = applicationInfo.labelRes
        return if (stringId == 0) applicationInfo.nonLocalizedLabel.toString() else context.getString(
            stringId
        )
    }
}
